### ISSUE TITLE
PropertyList.get uses valueOf, but this isn't defined for every type

### ISSUE THREAD

**Issue by kbuzby** [postmanlabs/postman-collection#608]
[PropertyList.get](http://www.postmanlabs.com/postman-collection/collection_property-list.js.html#line385) uses valueOf. However, not every type that is used in a PropertyList has a valueOf method (see Certificate). 

The valueOf method should be defined on the Property class so that a default implementation can be inherited by all properties, and then overridden by specific implementations.

**Comment by shamasis** [comment]
Property is too low level to define valueOf for everything.

@kamalaknn @harryi3t - thoughts?

Also, @kbuzby - I am curious what use of valueOf you are deriving other than that of in variables?

**Comment by kbuzby** [comment]
I have only been using it in variables. This was something I had just discovered when writing the TS type definitions, and was trying to make sure the definitions matched the actual and intended implementation as best as possible.

I mistakenly was thinking as I read the code that if this was getting called on a class like CertificateList where valueOf is not explicitly defined that this would cause an error. However, I did not realize valueOf was defined on the base JS Object prototype, which allows this to work with no issue, and I think works as intended (it just returns the object itself.

I'm ok if this issue is closed.
